src/new.tsx

```diff
@@ -93,16 +99,86 @@ function Charming() {
     }
   }, [data, points]);
 
+  React.useEffect(() => {
+    // Data is required to determine grapheme cluster breaks in pasted strings
+    if (!data) return;
+
+    const onKeyDown = (e: KeyboardEvent) => {
+      if (e.key == "/") {
+        openSearch();
+        e.preventDefault();
+      }
+    };
+
+    const onCopy = (e: ClipboardEvent) => {
+      // If the user is trying to copy text normally, don't prevent them from doing so
+      const selection = window.getSelection();
+      if (selection && selection.type == "Range") return;
+
+      // Need to use window.location.hash, not just location.hash, so that the closure captures
+      // window (and thus location.hash is updated) rather than capturing location (where hash would
+      // not be updated)
+      // Avoids a dependency on points which would cause the effect to re-run often
+      const points = getHashPoints(window.location.hash, [0]);
+
+      e.clipboardData?.setData("text/plain", pointsToString(points));
+      e.preventDefault();
```

Comment:
(ditto)

Reply:
resolved in 66878296c5b6ec61cd3c0f3c8c98cd3d4b7afb98 :3